Source/DafnyDriver/CliCompilation.cs

```diff
@@ -325,33 +337,44 @@ private List<ICanVerify> FilterCanVerifies(List<ICanVerify> canVerifies, out int
 
     var filterPosition = Options.Get(VerifyCommand.FilterPosition);
     if (filterPosition == null) {
-      line = null;
+      range = new LineRange(start, end);
       return canVerifies;
     }
 
-    var regex = new Regex(@"(.*)(?::(\d+))?", RegexOptions.RightToLeft);
+    var regex = new Regex(@"(.*)(?::(\d+)?(-?)(\d+)?)?", RegexOptions.RightToLeft);
```

Comment:
I'm pretty sure this is equivalent (I tested it on Regex 101) and has two less characters
```suggestion
    var regex = new Regex(@"(.*)(?::(\d*)(-?)(\d*))?", RegexOptions.RightToLeft);
```